sklearn/naive_bayes.py

```diff
@@ -1184,10 +1215,11 @@ def _update_cat_count(X_feature, Y, cat_count, n_classes):
                 cat_count[j, indices] += counts[indices]
 
         self.class_count_ += Y.sum(axis=0)
+        self._init_n_categories(X)
```

Comment:
I would call this `_validate_n_categories`

Reply:
and I would prefer the following signature:

```python
self.n_categories_ = self._validate_n_categories(X)
```

`_validate_n_categories` could be decorated as a `@staticmethod`. It will be more explicit when reading the code that we set a fitting attribute if it returns something in fit than setting it in a private method.